lib/spack/spack/config.py

```diff
@@ -808,13 +816,35 @@ def they_are(t):
 #
 # Settings for commands that modify configuration
 #
-def default_modify_scope():
+def default_modify_scope(section):
     """Return the config scope that commands should modify by default.
 
     Commands that modify configuration by default modify the *highest*
     priority scope.
     """
-    return spack.config.config.highest_precedence_scope().name
+    default_edit_scope = spack.config.get('config:default_edit_scope')
```

Comment:
FWIW. You could simplify lines 825-827 using:  `default_edit_scope = spack.config.get('config:default_edit_scope', 'user')`.